--- plasma-framework-5.33.0/autotests/CMakeLists.txt.omv~	2017-04-22 12:06:59.468657654 +0200
+++ plasma-framework-5.33.0/autotests/CMakeLists.txt	2017-04-22 12:08:22.657082717 +0200
@@ -18,6 +18,10 @@ if(KDE_PLATFORM_FEATURE_BINARY_COMPATIBL
     set(PLASMA_NO_PACKAGE_EXTRADATA TRUE)
 endif()
 
+include_directories(
+	${CMAKE_BINARY_DIR}/src/plasmaquick/KF5PlasmaQuick_autogen/include
+)
+
 MACRO(PLASMA_UNIT_TESTS)
        FOREACH(_testname ${ARGN})
                add_executable(${_testname} ${_testname}.cpp)
